### PR TITLE
 LI: provided RegExp language injection into string literals

### PR DESCRIPTION
**Pull request by vlad20012**
Highlight & analyze regular expressions

image

Related to #2231

### CODE REVIEW

File: src/test/kotlin/org/rust/lang/utils/RsEscapesUtilsTest.kt [hunk 1]
```diff
@@ -57,3 +62,47 @@ class UnescapeRsTest(private val input: String,
...
+            arrayOf("""a\"b""", "a\"b", intArrayOf(0,1,3,4), true),
+            arrayOf("""a\'b""", "a\'b", intArrayOf(0,1,3,4), true),
+            arrayOf("""a\tb""", "a\tb", intArrayOf(0,1,3,4), true),
+            arrayOf("""a\nb""", "a\nb", intArrayOf(0,1,3,4), true),
+            arrayOf("""a\rb""", "a\rb", intArrayOf(0,1,3,4), true),
+            arrayOf("""a\0b""", "a\u0000b", intArrayOf(0,1,3,4), true),
+            arrayOf("""a\\b""", "a\\b", intArrayOf(0,1,3,4), true),
+            arrayOf("""a\x01b""", "a\u0001b", intArrayOf(0,1,5,6), true),
+            arrayOf("""a\x7Fb""", "a\u007Fb", intArrayOf(0,1,5,6), true),
+//            arrayOf("""a\x80b""", "a", intArrayOf(0,1,2), false),
```
Comment (Undin) [inline src/test/kotlin/org/rust/lang/utils/RsEscapesUtilsTest.kt]: Let's add a comment that `"a\x80b"` is not valid rust string but now we parse it as valid string

File: src/main/kotlin/org/rust/ide/injected/RsRegExpInjector.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            ?.path?.reference?.resolve() as? RsFunction
+            ?: return false
+        return fn.name == "new" && fn.implTypeName == "Regex" && fn.containingCargoPackage?.normName == "regex"
+    }
+
+    /** RegexSet::new(&["...", "...", "..."]) */
+    private fun isRegexSetNew(context: RsLitExpr): Boolean {
+        // use this logic for now because we can't resolve `RegexSet::new` (looks like it's implemented by a macro)
+        val call = (context.parent?.parent?.parent?.parent as? RsCallExpr)?.expr as? RsPathExpr ?: return false
+        return call.path.text == "RegexSet::new"
```
Comment (Undin) [inline src/main/kotlin/org/rust/ide/injected/RsRegExpInjector.kt]: Let's add a check of crate name here too

Comment (Undin) [reply src/main/kotlin/org/rust/ide/injected/RsRegExpInjector.kt]: Oh, I've realized we can't do it without successful resolve

**Comment by vlad20012** [comment]
Rebased

**Pull request merged**